### ISSUE TITLE
How can I set TTL when saving with om?

### ISSUE THREAD

**Issue by shay-dahan** [redis/rueidis#383]
How can I set TTL when saving with om? I know redis should support TTL for records.

**Comment by rueian** [comment]
> How can I set TTL when saving with om? I know redis should support TTL for records.

Hi @shay-dahan,

The om interface doesn’t have that functionality currently.

**Comment by rueian** [comment]
Hi @shay-dahan,

Do you still need this feature?

**Comment by rueian** [comment]
Yes, please stay tuned. Do you have any comments on the proposed interfaces?

**Comment by shay-dahan** [comment]
The interface is ok. Also passing it as optional argument is good in my opinion to the normal save method.

**Comment by rueian** [comment]
Hi @shay-dahan,

After reconsidering the APIs, I now think adding new methods or adding new optional arguments to the existing methods may not be a good idea.

Either way, they may create inconsistent API usage and confusion about their effects. For example, users may question the effect of calling `Save()` for updating a record that was first created by using `SaveTTL()`. Will the second `Save()` call modify the TTL? Different users may want different behaviors.

Therefore, instead of introducing changes to the API interfaces, I suggest we should change the data mapping:

```go
type ExampleRecord struct {
	Key  string    `json:"key" redis:",key"`
	Ver  int64     `json:"ver" redis:",ver"`
	ExAt time.Time `json:"-"   redis:",exat"`
	// ...
}
```

Users will be able to set record expiration time to the new `exat` field without confusion.

**Comment by shay-dahan** [comment]
Makes sense to me because  the TTL is a property more than it is a method.

**Comment by shay-dahan** [comment]
@rueian hope you are doing well. Do you plan to implement the ExAt mechanism? I would love to assist if you'd like

**Comment by rueian** [comment]
Hi @shay-dahan,

It is still under development but almost finished https://github.com/redis/rueidis/pull/415. 

If you could give it a try, it would be helpful. Thanks!

**Comment by rueian** [comment]
Hi @shay-dahan,

This feature has been released in v1.0.23. Please let me know if it suits your needs.

**Comment by shay-dahan** [comment]
@rueian I am now giving it a try! Thank you so much!
